src/meta/src/stream/source_manager.rs

```diff
@@ -588,6 +588,32 @@ where
     )
 }
 
+pub fn validate_assignment(assignment: &mut HashMap<ActorId, Vec<SplitImpl>>) {
```

Comment:
> Please revert after resolving the doubled assignment issue.

As it's just an in-memory validation, I think it's totally okay to keep it here.